sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -489,6 +492,13 @@ private[sql] class JDBCRDD(
       }
       try {
         if (null != conn) {
+          if (!conn.getAutoCommit && !conn.isClosed) {
+            try {
+              conn.commit()
+            } catch {
+              case e: Exception => logWarning("Exception committing transaction", e)
```

Comment:
Was just thinking of using `finally`, and bringing the inner `if` statement inside the `try` to keep it tidy.